contracts/governance/GovernanceSettings.sol

```diff
@@ -3,51 +3,99 @@ pragma solidity 0.8.27;
 
 import {Decimal} from "../common/Decimal.sol";
 import {Governable} from "../model/Governable.sol";
-import {SoftwareUpgradeProposal} from "../proposal/SoftwareUpgradeProposal.sol";
-import {Constants} from "./Constants.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
 
 /// @notice GovernanceSettings is a contract for managing governance settings
-contract GovernanceSettings is Constants {
-    uint256 constant _proposalFee = _proposalBurntFee + _taskHandlingReward + _taskErasingReward;
-    uint256 constant _proposalBurntFee = 50 * 1e18;
-    uint256 constant _taskHandlingReward = 40 * 1e18;
-    uint256 constant _taskErasingReward = 10 * 1e18;
-    uint256 constant _maxOptions = 10;
-    uint256 constant _maxExecutionPeriod = 3 days;
+contract GovernanceSettings is Ownable {
+    /// @notice proposalFee is the fee of a proposal
+    uint256 public proposalFee;
+    /// @notice proposalBurntFee is the burnt part of the fee for a proposal
+    uint256 public proposalBurntFee;
+    /// @notice taskHandlingReward is a reward for handling each task
+    uint256 public taskHandlingReward;
+    /// @notice taskErasingReward is a reward for erasing each task
+    uint256 public taskErasingReward;
+    /// @notice maxOptions is the maximum number of options that can be offered in a single proposal
+    uint256 public maxOptions;
+    /// @notice maxExecutionPeriod is the period after the end of voting during which the proposal can be executed 
+    uint256 public maxExecutionPeriod;
+
+    // reverted when proposal fee would be under the sum of burnt fee and rewards
+    error ProposalFeeTooLow(uint256 necessaryValue);
 
-    /// @notice proposalFee is the fee for a proposal
-    /// @return proposal fee
-    function proposalFee() public pure returns (uint256) {
-        return _proposalFee;
+    constructor() Ownable(msg.sender) {
+        // Set default values
+        proposalBurntFee = 50 * 1e18;
+        taskHandlingReward = 40 * 1e18;
+        taskErasingReward = 10 * 1e18;
+        maxOptions = 10;
+        maxExecutionPeriod = 72 hours;
+        proposalFee = proposalBurntFee + taskHandlingReward + taskErasingReward;
     }
 
-    /// @notice proposalBurntFee is the burnt part of fee for a proposal
-    /// @return proposal burn fee
-    function proposalBurntFee() public pure returns (uint256) {
-        return _proposalBurntFee;
+    /// @notice setProposalFee sets the fee for a proposal
+    /// @param _proposalFee new proposal fee in wei of native tokens
+    function setProposalFee(uint256 _proposalFee) public onlyOwner {
+        uint256 necessarySum = proposalBurntFee + taskHandlingReward + taskErasingReward;
+        if(_proposalFee < necessarySum) {
+            revert ProposalFeeTooLow(necessarySum);
+        }
```

Comment:
please rename `necessarySum` to `minProposalFee` to be consistent with the same in following methods